src/server/api/routers/user.ts

```
@@ -22,4 +22,56 @@ export const userRouter = createTRPCRouter({
         })
       }
     }),
+  deleteManyUsers: protectedProcedure
+    .input(z.object({ ids: z.array(z.string()) }))
+    .mutation(async ({ input, ctx }) => {
+      const deleteUsers = await ctx.prisma.user.deleteMany({
+        where: {
+          id: {
+            in: input.ids,
+          },
+        },
+      })
+      if (!deleteUsers) {
+        throw new TRPCError({
+          code: 'INTERNAL_SERVER_ERROR',
+          message: 'Failed to delete users',
+        })
+      }
+    }),
+  updateUserEmailPreference: protectedProcedure
+    .input(z.object({ ids: z.array(z.string()) }))
+    .mutation(async ({ input, ctx }) => {
+      const promises = input.ids.map(async (_id) => {
+        const currentEmailVal = await ctx.prisma.user.findUnique({
+          where: {
+            id: _id,
+          },
+          select: {
+            shouldEmail: true,
+          },
+        })
```

Maybe throw an error here as well to check for `currentEmailVal`, then you won't need to worry about `currenEmailVal` being `null` in `user.update`.